scalanet/src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -107,6 +110,30 @@ object PeerGroup {
 
   abstract class TerminalPeerGroup[A, M](implicit codec: Codec[M]) extends PeerGroup[A, M]
 
+  abstract class PeerGroupWithProxySupport[A, M](implicit codec: Codec[M]) extends PeerGroup[A, M] {
```

Comment:
By the way I'm not sure what the significance of `TerminalPeerGroup` was but this could (as in the example I posted below) stay a trait, since the `DynamicTLSPeerGroup` implementing this feature is still a terminal one, while as I've shown this could be done with a non-terminal wrapper group.